From c520500225018c545bb23c9a4e2f514f26cf47f0 Mon Sep 17 00:00:00 2001
From: Cyril Hrubis <chrubis@suse.cz>
Date: Mon, 27 May 2013 14:00:27 +0200
Subject: [PATCH] openposix: generate-makefiles.sh: Fix buildonly tests.

The buildonly tests were not compiled due to mistake
in the generate-makefiles script. Sorry.

Signed-off-by: Cyril Hrubis <chrubis@suse.cz>
---
 testcases/open_posix_testsuite/scripts/generate-makefiles.sh | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/testcases/open_posix_testsuite/scripts/generate-makefiles.sh b/testcases/open_posix_testsuite/scripts/generate-makefiles.sh
index 9c758d2..b933033 100755
--- a/testcases/open_posix_testsuite/scripts/generate-makefiles.sh
+++ b/testcases/open_posix_testsuite/scripts/generate-makefiles.sh
@@ -36,6 +36,7 @@ generate_makefile() {
 	local make_copy_prereq_cache=
 	local prereq_cache=
 	local tests=
+	local targets=
 
 	local makefile=$1
 	local prereq_dir=$2
@@ -64,6 +65,15 @@ generate_makefile() {
 		fi
 
 		# Stuff that needs to be compiled.
+		if echo "$prereq" | grep -Eq '\.(run-test|sh|test)'; then
+			if [ "$targets" != "" ]; then
+				targets="$targets "
+			fi
+
+			targets="$targets${test_prefix}_$prereq"
+		fi
+
+		# Cache for generating compile rules.
 		case "$prereq" in
 		*.sh)
 			# Note that the sh scripts are copied later in order to
@@ -138,7 +148,7 @@ EOF
 
 	cat >> "$makefile.2" <<EOF
 INSTALL_TARGETS+=	${tests}
-MAKE_TARGETS+=		${tests}
+MAKE_TARGETS+=		${targets}
 
 EOF
 
-- 
1.8.1.5

